common-client/src/main/scala/org/genivi/sota/client/DeviceRegistryClient.scala

```diff
@@ -106,7 +106,9 @@ class DeviceRegistryClient(baseUri: Uri, devicesUri: Uri, deviceGroupsUri: Uri,
         case StatusCodes.NoContent if ct.runtimeClass == classOf[NoContent] =>
           FastFuture.successful(org.genivi.sota.client.NoContent()).asInstanceOf[Future[T]]
         case other if other.isSuccess() => unmarshaller(response.entity)
-        case err => FastFuture.failed(new Exception(err.toString))
+        case err => log.error(s"Got exception for request to ${httpRequest.uri}\n" +
+          s"Error message: ${response.entity.toString}")
```

Comment:
Did you try this locally?

I am not sure, but I think this will only give you the id of the jvm object for entity. You'll need to get the databytes and parse them to an utf string. Or maybe try just output response, I think response might try to do the decoding of it's entity when `toString` is called.

But I think this might work if you